test/form/index.js

```diff
@@ -0,0 +1,194 @@
+
+    instance.setValid(false);
+    assert.isFalse(instance.state.isValid, 'state was changed');
+
+    instance.addInput({ setPristine });
+    instance.setPristine(callback);
+
+    assert.deepEqual(instance.state, instance.getDefaultState(), 'state is back to default');
+    assert.isTrue(setPristine.calledOnce, 'setPristine on component was called');
+
+    process.nextTick(() => {
+      assert.isTrue(callback.calledOnce, 'callback was called');
+    });
+  });
+
+  it('setValid', () => {
+    const props = { onValidityChange: fake() };
+    const wrapper = shallow(createElement(Form, props));
+    const instance = wrapper.instance();
+    const doneFake = fake();
+
+    assert.isTrue(instance.state.isValid, 'default state is valild');
+
+    instance.setValid(false, doneFake);
+
+    assert.isTrue(doneFake.calledOnce, 'callback was called');
+    assert.isTrue(props.onValidityChange.calledOnce, 'onValidityChange prop was called');
+    assert.isFalse(instance.state.isValid, 'state was changed');
+  });
+
+  it('isDisabled', () => {
+    // this.props.locked is true
+    const props = { locked: true };
+    const wrapperWithLock = shallow(createElement(Form, props));
+    const instanceWithLock = wrapperWithLock.instance();
+    assert.isTrue(instanceWithLock.isDisabled(), 'form is disabled if locked is true');
+
+    // this.state.isValid is false
+    const wrapper = shallow(createElement(Form));
+    const instance = wrapper.instance();
+
+    instance.setValid(false);
+    assert.isTrue(instance.isDisabled(), 'form is disabled if state is not valid');
+  });
+
+  it('updateValidity', () => {
+    const wrapper = shallow(createElement(Form));
+    const instance = wrapper.instance();
+    instance.setValid = fake();
+
+    instance.updateValidity();
+
+    assert.isTrue(instance.setValid.calledOnce, 'setValid was called');
+  });
+
+  it('reset', () => {
+    const wrapper = shallow(createElement(Form));
+    const instance = wrapper.instance();
+    const reset = fake();
+
+    instance.addInput({ reset });
+    instance.setValid(false);
+    assert.isFalse(instance.state.isValid, 'state was changed');
+
+    instance.reset();
+    assert.deepEqual(instance.state, instance.getDefaultState(), 'state is back to default');
+    assert.isTrue(reset.calledOnce, 'reset on component was called');
+  });
+
+  it('submit', () => {
+    const props = {
+      onValidSubmit: fake(),
+      onInvalidSubmit: fake(),
+      onSubmit: fake(),
+    };
+    const preventDefault = fake();
+    const wrapper = shallow(createElement(Form, props));
+    const instance = wrapper.instance();
+    instance.validate = spy(() => Promise.resolve());
+
+    wrapper.find('form').simulate('submit', { preventDefault });
+
+    assert.isTrue(preventDefault.calledOnce, 'preventDefault was called');
+    assert.isTrue(instance.validate.calledOnce, 'validate was called');
+    assert.isTrue(props.onSubmit.calledOnce, 'onSubmit was called');
+
+    setTimeout(() => {
```

Comment:
Use `process.nextTick`, and this test is async, use `done`

Reply:
keeped setTimeout